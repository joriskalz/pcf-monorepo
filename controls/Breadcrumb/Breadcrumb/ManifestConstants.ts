export const enum ManifestPropertyNames {
    dataset = 'dataset',
}

export const enum ItemColumns {
    DisplayName = 'ItemDisplayName',
    Key = 'ItemKey',
    Clickable = 'ItemClickable',
}

export const enum InputEvents {
    SetFocus = 'SetFocus',
}

export const enum InputProperties {
    InputEvent = 'InputEvent',
    SelectedKey = 'SelectedKey',
}
