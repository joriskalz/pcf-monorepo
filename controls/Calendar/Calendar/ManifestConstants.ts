export const enum InputEvents {
    SetFocus = 'SetFocus',
}
