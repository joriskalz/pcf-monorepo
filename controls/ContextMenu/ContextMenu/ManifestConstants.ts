export const enum ManifestPropertyNames {
    dataset = 'dataset',
}

export const enum ItemColumns {
    DisplayName = 'ItemDisplayName',
    Key = 'ItemKey',
    Enabled = 'ItemEnabled',
    IconName = 'ItemIconName',
    IconColor = 'ItemIconColor',
    ParentKey = 'ItemParentKey',
    IconOnly = 'ItemIconOnly',
    Visible = 'ItemVisible',
    Checked = 'ItemChecked',
    ItemHeader = 'ItemHeader',
    ItemTopDivider = 'ItemTopDivider',
    ItemDivider = 'ItemDivider',
}

export const enum InputEvents {
    SetFocus = 'SetFocus',
}

export const enum InputProperties {
    InputEvent = 'InputEvent',
}

export const TextAlignmentTypes = {
    '0': 'center',
    '1': 'left',
    '2': 'right',
};
