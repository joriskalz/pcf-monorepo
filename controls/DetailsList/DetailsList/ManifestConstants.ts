export const enum RecordsColumns {
    RecordKey = 'RecordKey',
    RecordCanSelect = 'RecordCanSelect',
    RecordSelected = 'RecordSelected',
}

export const enum ColumnsColumns {
    ColDisplayName = 'ColDisplayName',
    ColName = 'ColName',
    ColWidth = 'ColWidth',
    ColSortable = 'ColSortable',
    ColSortBy = 'ColSortBy',
    ColHorizontalAlign = 'ColHorizontalAlign',
    ColVerticalAlign = 'ColVerticalAlign',
    ColMultiLine = 'ColMultiLine',
    ColIsBold = 'ColIsBold',
    ColTagColorColumn = 'ColTagColorColumn',
    ColTagBorderColorColumn = 'ColTagBorderColorColumn',
    ColResizable = 'ColResizable',
    ColHeaderPaddingLeft = 'ColHeaderPaddingLeft',
    ColCellType = 'ColCellType',
    ColShowAsSubTextOf = 'ColShowAsSubTextOf',
    ColPaddingTop = 'ColPaddingTop',
    ColPaddingLeft = 'ColPaddingLeft',
    ColLabelAbove = 'ColLabelAbove',
    ColMultiValueDelimiter = 'ColMultiValueDelimiter',
    ColFirstMultiValueBold = 'ColFirstMultiValueBold',
    ColInlineLabel = 'ColInlineLabel',
    ColHideWhenBlank = 'ColHideWhenBlank',
    ColSubTextRow = 'ColSubTextRow',
    ColAriaTextColumn = 'ColAriaTextColumn',
    ColCellActionDisabledColumn = 'ColCellActionDisabledColumn',
    ColImageWidth = 'ColImageWidth',
    ColImagePadding = 'ColImagePadding',
    ColRowHeader = 'ColRowHeader',
}

export const enum InputProperties {
    InputEvent = 'InputEvent',
}

export const enum InputEvents {
    SetFocus = 'SetFocus',
    SetFocusOnRow = 'SetFocusOnRow',
    SetFocusOnRowSetSelection = 'SetFocusOnRowSetSelection',
    ClearSelection = 'ClearSelection',
    SetSelection = 'SetSelection',
    SetFocusOnHeader = 'SetFocusOnHeader',
    LoadNextPage = 'LoadNextPage',
    LoadPreviousPage = 'LoadPreviousPage',
    LoadFirstPage = 'LoadFirstPage',
}

export const enum OutputEvents {
    Sort = 'Sort',
    CellAction = 'CellAction',
    OnRowSelectionChange = 'OnRowSelectionChange',
}

export const enum SortDirection {
    Ascending = '0',
    Descending = '1',
}

export const enum CellTypes {
    Expand = 'expand',
    Tag = 'tag',
    Image = 'image',
    ClickableImage = 'clickableimage',
    IndicatorTag = 'indicatortag',
    Link = 'link',
}
