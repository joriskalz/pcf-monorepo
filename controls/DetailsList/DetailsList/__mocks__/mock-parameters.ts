/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockEnumProperty, MockStringProperty, MockTwoOptionsProperty, MockWholeNumberProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        AccessibilityLabel: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        Theme: new MockStringProperty(),
        Compact: new MockTwoOptionsProperty(),
        CurrentSortColumn: new MockStringProperty(),
        CurrentSortDirection: new MockEnumProperty('0'),
        PageSize: new MockWholeNumberProperty(),
        RaiseOnRowSelectionChangeEvent: new MockTwoOptionsProperty(),
        SelectRowsOnFocus: new MockTwoOptionsProperty(),
        SelectionType: new MockEnumProperty('0'),
        AlternateRowColor: new MockStringProperty(),
        HeaderVisible: new MockTwoOptionsProperty(false),
        SelectionAlwaysVisible: new MockTwoOptionsProperty(false),
        records: new MockDataSet([]),
        columns: new MockDataSet([]),
        LargeDatasetPaging: new MockTwoOptionsProperty(),
    };
}
