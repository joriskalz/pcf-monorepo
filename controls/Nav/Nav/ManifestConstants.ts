export const enum ManifestPropertyNames {
    dataset = 'dataset',
}

export const enum ItemColumns {
    DisplayName = 'ItemDisplayName',
    Key = 'ItemKey',
    Enabled = 'ItemEnabled',
    IconName = 'ItemIconName',
    IconColor = 'ItemIconColor',
    TextColor = 'ItemTextColor',
    ParentKey = 'ItemParentKey',
    Visible = 'ItemVisible',
    Expanded = 'ItemExpanded',
}

export const enum InputEvents {
    SetFocus = 'SetFocus',
}

export const enum InputProperties {
    SelectedKey = 'SelectedKey',
}
