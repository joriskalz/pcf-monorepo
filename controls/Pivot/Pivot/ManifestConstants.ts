export const enum ManifestPropertyNames {
    dataset = 'dataset',
}

export const enum ItemColumns {
    DisplayName = 'ItemDisplayName',
    Key = 'ItemKey',
    Enabled = 'ItemEnabled',
    IconName = 'ItemIconName',
    IconColor = 'ItemIconColor',
    TextColor = 'ItemTextColor',
    ParentKey = 'ItemParentKey',
    IconOnly = 'ItemIconOnly',
    Visible = 'ItemVisible',
    Checked = 'ItemChecked',
    ItemCount = 'ItemCount',
}

export const enum InputEvents {
    SetFocus = 'SetFocus',
    Reset = 'Reset',
}

export const enum InputProperties {
    SelectedKey = 'SelectedKey',
}

export class RenderTypes {
    static PivotLinks = '0';
    static PivotTabs = '1';
}

export class RenderSize {
    static Normal = '0';
    static Compact = '1';
    static Large = '2';
}
