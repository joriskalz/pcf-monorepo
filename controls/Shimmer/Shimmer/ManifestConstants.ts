export const enum ManifestPropertyNames {
    dataset = 'dataset',
}

export const enum ItemColumns {
    Key = 'ItemKey',
    Type = 'ItemType',
    Height = 'ItemHeight',
    Width = 'ItemWidth',
    VerticalAlign = 'ItemVerticalAlign',
    RowKey = 'ItemRowKey',
}

export const enum RowColumns {
    Key = 'RowKey',
    Order = 'RowOrder',
    Count = 'RowCount',
    Width = 'RowWidth',
}

export const enum InputEvents {
    SetFocus = 'SetFocus',
    Reset = 'Reset',
}

export const enum InputProperties {
    SelectedKey = 'SelectedKey',
}

export class RenderTypes {
    static PivotLinks = '0';
    static PivotTabs = '1';
}

export class RenderSize {
    static Normal = '0';
    static Compact = '1';
    static Large = '2';
}
