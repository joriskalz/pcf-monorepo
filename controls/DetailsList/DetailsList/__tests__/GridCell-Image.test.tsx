import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { IGridColumn } from '../Component.types';
import { GridCell, GridCellProps } from '../GridCell';
import { CellTypes } from '../ManifestConstants';
import { MockEntityRecord } from '../__mocks__/mock-datasets';

const item = new MockEntityRecord('1', {
    a: 'https://via.placeholder.com/300x200',
    b: 'icon:Save',
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
} as any);

const defaultProps: GridCellProps = {
    item: item,
    index: 1,
    column: {
        key: 'a',
        name: 'a',
        fieldName: 'a',
        minWidth: 200,
        maxWidth: 400,
        calculatedWidth: 200,
        isResizable: true,
        isSorted: false,
        cellType: CellTypes.Image,
    } as IGridColumn,
    onCellAction: () => {
        //noop
    },
};

describe('GridCell', () => {
    it('image cell', () => {
        const mockProps: GridCellProps = {
            ...defaultProps,
            column: { ...defaultProps.column, fieldName: 'a' } as IGridColumn,
        };
        const component = renderer.create(<GridCell {...mockProps} />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('icon cell', () => {
        const mockProps: GridCellProps = {
            ...defaultProps,
            column: { ...defaultProps.column, fieldName: 'b' } as IGridColumn,
        };

        const component = renderer.create(<GridCell {...mockProps} />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('icon cell align center', () => {
        const mockProps: GridCellProps = {
            ...defaultProps,
            column: { ...defaultProps.column, fieldName: 'b', verticalAligned: 'center' } as IGridColumn,
        };

        const component = renderer.create(<GridCell {...mockProps} />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
