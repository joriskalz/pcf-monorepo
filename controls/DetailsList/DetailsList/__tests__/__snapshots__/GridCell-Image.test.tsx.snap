// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridCell icon cell 1`] = `
<div
  className=

      {
        display: flex;
        flex-direction: row;
        flex-wrap: wrap;
      }
>
  <span
    className=

        {
          font-weight: 400;
          max-width: 400px;
          overflow: hidden;
          text-overflow: ellipsis;
          white-space: nowrap;
        }
  >
    <div
      className=

          {
            display: flex;
            height: 100%;
          }
      title=""
    >
      <i
        aria-hidden="true"
        className=
            ms-Icon
            {
              display: inline-block;
            }
            {
              color:  !important;
              font-size: 18px;
            }
        data-icon-name="Save"
        style={
          Object {
            "fontFamily": undefined,
          }
        }
      />
    </div>
  </span>
</div>
`;

exports[`GridCell icon cell align center 1`] = `
<div
  className=

      {
        align-items: center;
        display: flex;
        flex-direction: row;
        flex-wrap: wrap;
        height: 100%;
      }
>
  <span
    className=

        {
          font-weight: 400;
          max-width: 400px;
          overflow: hidden;
          text-overflow: ellipsis;
          white-space: nowrap;
        }
  >
    <div
      className=

          {
            display: flex;
            height: 100%;
          }
      title=""
    >
      <i
        aria-hidden="true"
        className=
            ms-Icon
            {
              display: inline-block;
            }
            {
              color:  !important;
              font-size: 18px;
            }
        data-icon-name="Save"
        style={
          Object {
            "fontFamily": undefined,
          }
        }
      />
    </div>
  </span>
</div>
`;

exports[`GridCell image cell 1`] = `
<div
  className=

      {
        display: flex;
        flex-direction: row;
        flex-wrap: wrap;
      }
>
  <span
    className=

        {
          font-weight: 400;
          max-width: 400px;
          overflow: hidden;
          text-overflow: ellipsis;
          white-space: nowrap;
        }
  >
    <div
      className=

          {
            display: flex;
            height: 100%;
          }
      title=""
    >
      <div
        className=
            ms-Image
            {
              -moz-osx-font-smoothing: grayscale;
              -webkit-font-smoothing: antialiased;
              font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
              font-size: 14px;
              font-weight: 400;
              overflow: hidden;
            }
        style={
          Object {
            "height": undefined,
            "width": 32,
          }
        }
      >
        <img
          className=
              ms-Image-image
              ms-Image-image--portrait
              is-notLoaded
              is-fadeIn
              {
                display: block;
                height: auto;
                opacity: 0;
                width: 100%;
              }
          onError={[Function]}
          onLoad={[Function]}
          src="https://via.placeholder.com/300x200"
        />
      </div>
    </div>
  </span>
</div>
`;
